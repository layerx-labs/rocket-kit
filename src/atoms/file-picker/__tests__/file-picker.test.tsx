import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { FilePicker } from '../../..';

describe('FilePicker', () => {
  it('renders', () => {
    const { asFragment } = render(
      <FilePicker name="file" onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
