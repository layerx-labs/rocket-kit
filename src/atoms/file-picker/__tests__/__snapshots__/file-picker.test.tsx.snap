// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePicker renders 1`] = `
<DocumentFragment>
  .c0 input {
  width: 0.1px;
  height: 0.1px;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
}

.c0 label {
  margin-bottom: 0 !important;
  border-width: 0.0625rem;
  border-style: solid;
  border-color: #D9D4ED;
  border-radius: 0.5rem;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 3.125rem;
  -webkit-appearance: none;
  outline: none;
  cursor: pointer;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  overflow: hidden;
}

.c0 label span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 label span.file-name {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: block;
  padding: 0 0.625rem;
  color: #000000;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0 label span.button {
  border-left: 0.0625rem solid #D9D4ED;
  background-color: #D9D4ED;
  height: 100%;
  padding: 0 1.25rem;
  color: #4329A6;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 label span.button svg {
  margin-right: 0.5rem;
  width: auto;
  height: 1.5rem;
  fill: #4329A6;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 label span.button:hover {
  background-color: #C7BFE4;
}

.c0 label:hover {
  border-color: #C7BFE4;
}

.c0 label:hover span.button {
  background-color: #C7BFE4;
}

<div
    class="c0"
  >
    <input
      data-multiple-caption="Select file"
      id="file"
      type="file"
    />
    <label
      for="file"
    >
      <span
        class="file-name"
      >
        Select file
      </span>
      <span
        class="button"
      >
        <svg
          class="icon"
          viewBox="0 0 32 32"
        >
          <path
            d="M25.8 13.387c-0.907-4.6-4.947-8.053-9.8-8.053-3.853 0-7.2 2.187-8.867 5.387-4.013 0.427-7.133 3.827-7.133 7.947 0 4.413 3.587 8 8 8h17.333c3.68 0 6.667-2.987 6.667-6.667 0-3.52-2.733-6.373-6.2-6.613zM18.667 17.333v5.333h-5.333v-5.333h-4l6.667-6.667 6.667 6.667h-4z"
          />
        </svg>
        Browse
      </span>
    </label>
  </div>
</DocumentFragment>
`;
